### PR TITLE
Add an after_bundle callback in Rails plugin templates

### LINKED ISSUE #23315: Rails 4.2 added support for after_bundle to app generation but not for plugin generation
#16359 added support for after_bundle to app generation, but did not for plugin generation.  This prevents certain actions from occurring that may be dependent upon bundle install (e.g. rspec test generation).  It is somewhat misleading as the developer guide makes it seem like after_bundle should work in both kinds of generation.

### PR DESCRIPTION
**Pull request by rfmanuel**
Closes #23315.  This is based heavily off of the pull request at #16359.  after_bundle was added as part of that to give a hook to execute actions after bundle install was executed during app generation but not for plugin generation.  This pull request adds that functionality to plugin generation as well.

### CODE REVIEW

File: railties/CHANGELOG.md [hunk 1]
```diff
@@ -448,4 +448,11 @@
 
     *Alex Robbin*
 
+*   Add `after_bundle` callbacks in Rails plugin templates.  Useful for allowing
+    templates to perform actions that are dependent upon `bundle install`.
+
+    Fixes #<23315>
```
Comment (rafaelfranca) [inline railties/CHANGELOG.md]: `Fixes #23315`

File: railties/CHANGELOG.md [hunk 2]
```diff
@@ -448,4 +448,11 @@
 
     *Alex Robbin*
 
+*   Add `after_bundle` callbacks in Rails plugin templates.  Useful for allowing
```
Comment (rafaelfranca) [inline railties/CHANGELOG.md]: New entries are in the top of the file

File: guides/source/5_0_release_notes.md [hunk 3]
```diff
@@ -120,6 +120,9 @@ Please refer to the [Changelog][railties] for detailed changes.
     ([Pull Request](https://github.com/rails/rails/pull/22457),
      [Pull Request](https://github.com/rails/rails/pull/22288))
 
+*   Introduced an `after_bundle` callback for use in Rails plugin templates.
```
Comment (rafaelfranca) [inline guides/source/5_0_release_notes.md]: We can remove this from here. It is not a kind of feature that I'd give highlight.

**Comment by rfmanuel** [comment]
I went ahead and made the requested changes

**Pull request merged**